### ISSUE TITLE
Performance gap between Wezterm and Windows Terminal on Windows 11 VM

### ISSUE THREAD

**Issue by tamlok** [wez/wezterm#4700]
### What Operating System(s) are you seeing this problem on?

Windows

### WezTerm version

20231203-124028-e3cd2e93

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

I am doing daily dev work via remote desktop to an azure VM on Win11. When using NVIM within Wezterm, I can notice poor rendering perf (such as when moving cursor down repeatedly). But in Windows Terminal, the rendering feels as good as native desktop in NVIM.

How can I help to provide more info to improve the perf and minimize the gap?

### To Reproduce

Create a Win11 VM and remote desktop to it and then launch NVIM and keep pressing J while opening a medium size file.

### Configuration

    front_end = "WebGpu",

### Expected Behavior

Should behave with the same perf as Windows Terminal.

### Logs

Debug Overlay
wezterm version: 20231203-124028-e3cd2e93 x86_64-pc-windows-msvc
Window Environment: Windows
WebGPU: name=Microsoft Basic Render Driver, device_type=Cpu, backend=Dx12, vendor=5140, device=140

**Comment by wez** [comment]
It looks like you're using Cpu rendering, so the performance will not be good.
For remote work, I recommend using either ssh or wezterm's multiplexing feature so that your local machine can render the gui.

**Comment by tamlok** [comment]
Is there any command to check Windows Terminal's rendering type? If both of them are CPU rendering, then there must be something different elsewhere?

**Comment by tamlok** [comment]
Hi @wez , I read the docs about font_end and see this:

```
WezTerm will automatically select Software if it detects that it is being started in a Remote Desktop environment on Windows.
```

So WezTerm will use Software if it is in a remote desktop session regardless of the config?

I did a recording comparing WezTerm with Windows Terminal. I type the chars wihtout any hesitation bug WezTerm responses very slow and suddenly display the whole sentence I typed.


https://github.com/wez/wezterm/assets/5763157/bc0c6c79-5337-431d-82de-f00d5de3859e

**Comment by DnBaal** [comment]
You can check windows terminal rendering by going in Settings -> Rendering.
And you can check available WebGpu in wezterm by going into logs, (ctrl-shift-L) and using print(wezterm.gui.enumerate_gpus())

I had same problem with performance , but on local machine. The problem was that wezterm choose Vulkan instead of Dx12 by default, so i had to use lua code to force it to use backend=Dx12, device_type=DiscreteGpu, and performance became just the same as windows terminal.

**Comment by tamlok** [comment]
Hi @DnBaal ,

Thanks for your help!

From the remote desktop, it seems that there is only CPU renderring:

```
wezterm version: 20241129-152148-4906789a x86_64-pc-windows-msvc
Window Environment: Windows
Lua Version: Lua 5.4
WebGPU: name=Microsoft Basic Render Driver, device_type=Cpu, backend=Dx12, driver=10.0.26100.3624, vendor=5140, device=140
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
> print(wezterm.gui.enumerate_gpus())
21:34:22.761 INFO logging > lua: [
    {
        "backend": "Dx12",
        "device": 140,
        "device_type": "Cpu",
        "driver": "10.0.26100.3624",
        "name": "Microsoft Basic Render Driver",
        "vendor": 5140,
    },
]
```

For Windows Terminal:

![Image](https://github.com/user-attachments/assets/f1aa5007-1b81-4647-bbe4-aa9e9e755112)

**Comment by sebastianstudniczek** [comment]
@DnBaal Would you mind sharing lua code you had to write in order to fix it?

**Comment by wez** [comment]
@sebastianstudniczek see https://wezterm.org/config/lua/config/webgpu_preferred_adapter.html

**Comment by sebastianstudniczek** [comment]
Thanks, I knew about this one but forgot that it's possible to enumerate available gpus and filter them in config.

Nevertheless trying different discrete gpu settings (Intel Arc A370M either on Vulkan or Dx12) didn't help. Still seeing reduced performance compared to Windows Terminal in NeoVim when scrolling for example. 
Also interestingly Mica effect stopped working when switched to Arc 370 with Dx12 but maybe that's issue with drivers. 

Note that I am using it on "real" hardware, not it in VM. Will try to tinker with other settings and maybe find something related

**Comment by DnBaal** [comment]
@sebastianstudniczek Did you try nightly build ?

**Comment by sebastianstudniczek** [comment]
Yes, it's better but still not as smooth as windows terminal

**Comment by DnBaal** [comment]
@sebastianstudniczek i think windows terminal anyway will be smoother , because its rooted deeply in windows bloat))

**Comment by sebastianstudniczek** [comment]
@DnBaal True, I would expect windows native terminal to work better then cross platform one. 

On the other hand I've also tried alacritty and it's is working as smooth as windows terminal, but guess it's not a fair comparison with wezterm given how featurerich wezterm is and alacritty being really minimal

**Comment by tamlok** [comment]
I use VibeCoding to add GDI support for Wezterm. It turns out to improve the perf within a RDP session hugely and now is on-par with Windows Terminal.

Is there a plan to support GDI front end? @wez 

https://github.com/wezterm/wezterm/pull/7951
